### PR TITLE
[genesis][breaking] un-revert previous attempt to integrate discovery in genesis

### PR DESCRIPTION
**Pull request by phlip9**

### CODE REVIEW

File: config/config-builder/src/full_node_config.rs [hunk 1]
```diff
@@ -167,7 +167,13 @@ impl FullNodeConfig {
...
         );
 
-        let (validator_configs, faucet_key) = self.validator_config.build_swarm()?;
+        // Don't randomize ports. Until we better separate genesis generation,
+        // we don't want to accidentally randomize initial discovery set addresses.
+        let randomize_service_ports = false;
+        let randomize_libranet_ports = false;
+        let (validator_configs, faucet_key) = self
+            .validator_config
+            .build_common(randomize_service_ports, randomize_libranet_ports)?;
```
Comment (phlip9) [inline config/config-builder/src/full_node_config.rs:176]: note for reviewer: this is the fix

Comment (davidiw) [reply config/config-builder/src/full_node_config.rs:176]: why is it okay for these to be non-randomized but the validator can be?

**Comment by phlip9** [comment]
> looks good besides my confusion

the validator genesis config for CT is also not randomized currently. talking with @ankushagarwal tomorrow about actually implementing this.

see 

1. https://github.com/libra/libra/blob/master/docker/validator-dynamic/docker-run-dynamic.sh#L36
2. https://github.com/libra/libra/blob/master/config/config-builder/src/main.rs#L297
3. https://github.com/libra/libra/blob/master/config/config-builder/src/validator_config.rs#L130
4. https://github.com/libra/libra/pull/3294/files#diff-1625684a48bef45fe806419650a0ac31R156

so the validator (and now full node) configs are just using placeholders in CT

File: config/config-builder/src/validator_config.rs [hunk 2]
```diff
@@ -153,7 +153,7 @@ impl ValidatorConfig {
     }
 
     pub fn build_set(&self) -> Result<Vec<NodeConfig>> {
-        let (configs, _) = self.build_common(false)?;
+        let (configs, _) = self.build_common(false, false)?;
```
Comment (davidiw) [inline config/config-builder/src/validator_config.rs:156]: this is why

**Pull request closed without merging**